File: assignments/two_sum/python/two_sum_test_runner.py
import sys
from two_sum import TwoSum

def run_tests():
    """
    Runs a series of test cases against the student's TwoSum implementation.
    """
    solver = TwoSum()

    test_cases = [
        ([2, 7, 11, 15], 9, [0, 1]),
        ([3, 2, 4], 6, [1, 2]),
        ([3, 3], 6, [0, 1]),
    ]

    passed_count = 0
    report = []

    print("===== Starting Automated Tests for TwoSum (Python) =====")

    for i, (nums_input, target_input, expected_output) in enumerate(test_cases):
        try:
            actual_output = solver.solve(nums_input, target_input)

            if actual_output:
                actual_output.sort()
            expected_output.sort()

            report.append(f"Test Case #{i + 1}: Input nums={nums_input}, target={target_input}")
            report.append(f" -> Expected: {expected_output}, Got: {actual_output}")

            if actual_output == expected_output:
                report.append(" -> ✅ PASS\n")
                passed_count += 1
            else:
                report.append(" -> ❌ FAIL\n")

        except Exception as e:
            report.append(f"Test Case #{i + 1}: Input nums={nums_input}, target={target_input}")
            report.append(f" -> ❌ ERROR: An exception occurred - {type(e).__name__}\n")

    print("\n".join(report))
    print("===== Test Summary =====")
    print(f"Total Tests: {len(test_cases)}")
    print(f"Passed: {passed_count}")
    print(f"Failed: {len(test_cases) - passed_count}")
    print("========================")

    print(f"[SUMMARY]:{passed_count}:{len(test_cases)}")

if __name__ == "__main__":
    sys.path.append('.')
    run_tests()

File: assignments/two_sum/javascript/two_sum_test_runner.js
const assert = require('assert');
const { TwoSum } = require('./TwoSum.js');

function runTests() {
    const solver = new TwoSum();
    const testCases = [
        { nums: [2, 7, 11, 15], target: 9, expected: [0, 1] },
        { nums: [3, 2, 4], target: 6, expected: [1, 2] },
        { nums: [3, 3], target: 6, expected: [0, 1] },
    ];

    let passedCount = 0;
    let report = [];
    console.log("===== Starting Automated Tests for TwoSum (JavaScript) =====");

    testCases.forEach((tc, i) => {
        let actual, expected;
        try {
            actual = solver.solve(tc.nums, tc.target);
            expected = tc.expected;

            actual.sort((a, b) => a - b);
            expected.sort((a, b) => a - b);

            assert.deepStrictEqual(actual, expected);
            report.push(`Test Case #${i + 1}: Input nums=${JSON.stringify(tc.nums)}, target=${tc.target} -> ✅ PASS`);
            passedCount++;
        } catch (error) {
            report.push(`Test Case #${i + 1}: Input nums=${JSON.stringify(tc.nums)}, target=${tc.target} -> ❌ FAIL/ERROR: Expected ${JSON.stringify(expected)}, but got ${JSON.stringify(actual)}`);
        }
    });

    console.log(report.join('\n\n'));
    console.log("\n===== Test Summary =====");
    console.log(`Total Tests: ${testCases.length}`); console.log(`Passed: ${passedCount}`); console.log(`Failed: ${testCases.length - passedCount}`); console.log("========================");
    console.log(`[SUMMARY]:${passedCount}:${testCases.length}`);
}

runTests();